### ISSUE TITLE
Collection list throws error

### ISSUE THREAD

**Issue by rs0h** [arangodb/arangojs#443]
Consumer collection has 1 document, but when I am trying to retrieve it with a code:

```
db.collection('Consumer').list().then(
      function (list) {console.log(list);},
      function (err) {console.error('Something went wrong:', err);}
    );

```
it produces an error:
Something went wrong: [ 'Consumer/181915' ]

Doing the same using query is working for me:

```
  db.query('FOR x IN Consumer RETURN x')
    .then(function (cursor) {return cursor.all();})
    .then(
      function (list) {console.log(list);},
      function (err) {console.error('Something went wrong:', err);}
    );
```

`[ { _key: '181915', _id: 'Consumer/181915', _rev: '_U3tYQiG---' } ]`

**Comment by kvahed** [comment]
`db.collection('Consumer').toArray()` is what you are looking for?

**Comment by rs0h** [comment]
Very likely to that, but it produces error also
```
500: Internal Server Error :: TypeError: db.collection(...).toArray is not a function
```

**Comment by jsteemann** [comment]
`db.collection(...).toArray()` does not exist in arangojs.
Calling `db.collection(...).list()` is appropriate.
However, looking into arangojs code I found that `list` behaves differently than other functions as it seems to execute its error callback on success.
So this command should work but doesn't:
```js
db.collection('Consumer').list().then( function(list) { console.log(list); });
```
And this one works but shouldn't:
```js
db.collection('Consumer').list().then( function(list) { console.log(list); }, function(list) { console.error("oops:", list); });
```
The following patch for arangojs seems to fix this:
```diff
--- lib/collection.js	2017-04-26 08:50:49.767889071 +0200
+++ lib/collection-new.js	2017-04-26 08:51:16.500394052 +0200
@@ -328,7 +328,7 @@
         });
       } else {
         this._api.put('/simple/all-keys', { type: type, collection: this.name }, function (err, res) {
-          return err ? callback(err) : callback(res.body.result);
+          return err ? callback(err) : callback(null, res.body.result);
         });
       }
       return promise;
```
@pluma?

**Comment by pluma** [comment]
@jsteemann well spotted! That should fix it.

**Comment by jsteemann** [comment]
@pluma: can you fix this in arangojs?

**Comment by pluma** [comment]
Fixed via 3e10eef.
